src/overlays/actors/ovl_Dm_Char04/z_dm_char04.c

```diff
@@ -28,18 +30,125 @@ const ActorInit Dm_Char04_InitVars = {
+
+        if (play->csCtx.frames == play->csCtx.actorActions[actionIndex]->startFrame) {
+            if (this->csAction != play->csCtx.actorActions[actionIndex]->action) {
+                this->csAction = play->csCtx.actorActions[actionIndex]->action;
+                if (play->csCtx.actorActions[actionIndex]->action == 1) {
+                    this->animationIndex = 0;
+                } else {
+                    this->animationIndex = 0;
+                }
+                func_80AABC40(&this->skelAnime, &sAnimations[this->animationIndex], 0);
+            }
+        }
+        Cutscene_ActorTranslateAndYaw(&this->actor, play, actionIndex);
+    } else {
+        this->csAction = 0x63;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char04/func_80AABC40.s")
+void DmChar04_Update(Actor* thisx, PlayState* play) {
+    DmChar04* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char04/DmChar04_Init.s")
+    SkelAnime_Update(&this->skelAnime);
+    this->actionFunc(this, play);
+    this->unk_262++;
+}
+static Vec3f D_80AAC4F0 = { 0.0f, 0.0f, 0.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char04/DmChar04_Destroy.s")
+s32 DmChar04_OverrideLimbDraw(PlayState* play2, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx,
+                              Gfx** gfx) {
+    PlayState* play = play2;
+    f32 sp28;
+    Vec3f sp1C;
+    DmChar04* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char04/func_80AABE34.s")
+    if (limbIndex == 6) {
+        sp28 = ((Math_SinS((this->unk_262 << 0xC)) * 0.1f) + 1.0f) * 0.012f * (this->actor.scale.x * 124.99999f);
+        Matrix_MultVec3f(&D_80AAC4F0, &sp1C);
+        Matrix_Translate(sp1C.x, sp1C.y, sp1C.z, MTXMODE_NEW);
+        Matrix_Scale(sp28, sp28, sp28, MTXMODE_APPLY);
+    }
+    return false;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char04/DmChar04_Update.s")
+void DmChar04_Draw(Actor* thisx, PlayState* play) {
+    Gfx* gfx = GRAPH_ALLOC(play->state.gfxCtx, sizeof(Gfx) * 4);
+    s32 phi_a0;
+    s32 pad;
+    DmChar04* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char04/func_80AABF74.s")
+    OPEN_DISPS(play->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char04/DmChar04_Draw.s")
+    func_8012C94C(play->state.gfxCtx);
+    phi_a0 = (this->unk_262 * 0x32) & 0x1FF;
+    if (phi_a0 >= 0x100) {
+        phi_a0 = 0x1FF - phi_a0;
+    }
+    gSPSegment(POLY_XLU_DISP++, 0x08, &gfx[0]);
+    gDPPipeSync(gfx++);
+    gDPSetPrimColor(gfx++, 0, 0x01, (u8)this->primColors.r, (u8)this->primColors.g, (u8)this->primColors.b,
+                    (u8)(this->primColors.a * 1));
+    gDPSetRenderMode(gfx++, G_RM_PASS, G_RM_ZB_CLD_SURF2);
+    gSPEndDisplayList(gfx);
+    gDPSetEnvColor(POLY_XLU_DISP++, (u8)(s8)this->envColors.r, (u8)(s8)this->envColors.g, (u8)(s8)this->envColors.b,
+                   (u8)(s8)((f32)phi_a0 * 1));
```

Comment:
Are the `s8` casts required?
```suggestion
    gDPSetEnvColor(POLY_XLU_DISP++, (u8)this->envColors.r, (u8)this->envColors.g, (u8)this->envColors.b,
                   (u8)phi_a0);
```

Reply:
doesn't match, please verify in local before reviewing